ReactSkia/views/common/RSkDrawUtils.cpp

```diff
@@ -301,43 +335,88 @@ void drawBorder(SkCanvas *canvas,
-             drawEdges(LeftEdge,canvas,frame,borderProps,backgroundColor,opacity);
+             drawEdges(LeftEdge,canvas,frame,borderProps,opacity);
          }
          /*Draw Top Side*/
          if((backgroundColor != borderProps.borderColors.top) && \
                  (isDrawVisible(borderProps.borderColors.top,opacity,borderProps.borderWidths.top) )){
-             drawEdges(TopEdge,canvas,frame,borderProps,backgroundColor,opacity);
+             drawEdges(TopEdge,canvas,frame,borderProps,opacity);
          }
          /*Draw Bottom Side*/
          if((backgroundColor != borderProps.borderColors.bottom) && \
                  (isDrawVisible(borderProps.borderColors.bottom,opacity,borderProps.borderWidths.bottom))){
-             drawEdges(BottomEdge,canvas,frame,borderProps,backgroundColor,opacity);
+             drawEdges(BottomEdge,canvas,frame,borderProps,opacity);
          }
     }
 }
-void  drawShadow(SkCanvas *canvas, 
-                               Rect frame,
-                               BorderMetrics borderProps,
-                               ShadowMetrics shadowMetrics)
-{
- 
-    if(isDrawVisible(shadowMetrics.shadowColor,shadowMetrics.shadowOpacity)){
-/*+ve shadowOffset.width  => move shadow towards Right, -ve => move shadow towards left
-  +ve shadowOffset.height => move shadow towards bottom, -ve => mov shadow towards top
- */
-         frame.origin.x= frame.origin.x+shadowMetrics.shadowOffset.width;
-         frame.origin.y=frame.origin.y+shadowMetrics.shadowOffset.height;
-/*Shadow effect acheived by drawing background with Blur effect*/
-         SkPaint paint;
-         paint.setMaskFilter(SkMaskFilter::MakeBlur(kNormal_SkBlurStyle, shadowMetrics.shadowRadius));
-         drawRect(Background,canvas,frame,borderProps,shadowMetrics.shadowColor,shadowMetrics.shadowOpacity,&paint);
+bool  drawShadow(SkCanvas *canvas,Rect frame,
+                        BorderMetrics borderProps,
+                        SharedColor backgroundColor,
+                        Float shadowOpacity,
+                        sk_sp<SkImageFilter> shadowFilter) {
+
+    if(shadowFilter == NULL) return false;
+    SkPaint paint;
+    paint.setImageFilter( shadowFilter);
+
+    DrawMethod shadowOn = None;
+    /*Shadow on Background : If bg  visible */
+    if(backgroundColor && isDrawVisible(backgroundColor)) {
+        shadowOn=Background;
     }
-} 
+    /*Shadow on Border : None of the Border width to be '0' and border color not transparent */
+    else if(borderProps.borderColors.isUniform() && isDrawVisible(borderProps.borderColors.left) \
+        && (borderProps.borderWidths.left !=0) && (borderProps.borderWidths.top !=0)  \
+        && (borderProps.borderWidths.right !=0) && (borderProps.borderWidths.bottom !=0)) {
+
+           shadowOn = Border;
+    }
+    if(shadowOn != None) {
+        canvas->saveLayerAlpha(NULL,shadowOpacity);
```

Comment:
Apply save Layer & restore only if opacity != 255.999

Reply:
Suggest to add inline function in Rskconversion.h, to get is Fully opaque, as it is used in more one place & files